richtext/plugins/org.polarsys.kitalpha.richtext.widget/src/org/polarsys/kitalpha/richtext/widget/internal/ListenerInstaller.java

```diff
@@ -236,44 +243,42 @@ protected void installChangeNotificationHandlerListener(final MDENebulaBasedRich
 	}
 
 	/**
-	 * Listener that temporary saves the editor content when receiving a 'change'
-	 * event.
-	 * 
-	 * @param widget the rich text widget.
-	 */
+   * Listener that is called when the content of the editor changes.<br/>
+   * It temporarily saves the editor content when receiving the 'change' event the first time.
+   */
 	protected void createChangeContentListener(final MDENebulaBasedRichTextWidget widget) {
-		new BrowserFunction(widget.getBrowser(), "firePropertyChangeEvent") { //$NON-NLS-1$
-			@Override
-			public Object function(Object[] arguments) {
-				if (!widget.isDirtyStateUpdated() && widget.isDirty()) {
-					// As a performance improvement, the saveContent is only
-					// called only if the dirty state of the widget is not
-					// updated
-					widget.saveContent();
-					widget.setDirtyStateUpdated(true);
-				}
-				return null;
-			}
-		};
-	}
-
-	protected void installChangeContentListener(final MDENebulaBasedRichTextWidget widget) {
-		StringBuilder script = new StringBuilder();
-
-		/**
-		 * Notice that firePropertyChangeEvent() javascript function is defined
-		 * MDERichTextEditor.
-		 */
-		script.append("CKEDITOR.instances.editor.on('key', function () {");
-		script.append("firePropertyChangeEvent();");
-		script.append("});");
-
-		if (!widget.executeScript(script.toString())) {
-			Activator.getDefault().getLog().log(new Status(IStatus.WARNING, Activator.PLUGIN_ID,
-					"Rich text widget cannot install firePropertyChangeEvent handler")); //$NON-NLS-1$
-		}
+    new BrowserFunction(widget.getBrowser(), "onChangeEvent") { //$NON-NLS-1$
+      @Override
+      public Object function(Object[] arguments) {
+        if (!widget.isDirtyStateUpdated()) {
+          widget.saveContent();
+          widget.setDirtyStateUpdated(true);
+        }
+        return null;
+      }
```

Comment:
Suppr/Del key pressed when cursor is at the end of the text triggers this function with no change (as backspace at first position), we must set the dirtyStateUpdate to true only if we now that the widget text has been changed, ie Widget text has been saved to the model.

Another  way would be to test the isDirty after the saveContent but some resourceStatus might be done async.

```
            public Object function(Object[] arguments) {
                if (!widget.isDirtyStateUpdated()) {
                    final boolean[] contentSavedAndChanged = { false };
                    widget.addPropertyChangeListener(evt -> {
                        if (evt.getSource() == widget && AbstractMDERichTextWidget.WIDGET_SAVED_PROP.equals(evt.getPropertyName())) {
                            contentSavedAndChanged[0] = true;
                        }
                    });

                    widget.saveContent();

                    if (contentSavedAndChanged[0]) {
                        widget.setDirtyStateUpdated(true);
                    }
                }
                return null;
            }
    ```

Reply:
The provided patch set works fine on Windows and macOS.